common-moonmage.lic

```diff
@@ -172,13 +173,21 @@ module DRCMM
     # Look at yourself to see if you're wearing a moon weapon.
     # This regex specifically checks for the messaging that it's in the air next to you
     # so that we avoid false positives like wearing a "black moonsilk shirt".
-    look_result = DRC.bput("look #{Char.name}", moon_floating_regex, "You are wearing")
+    # Known caveat, this won't work if you're wearing a shadowsilk cloak or something that hides all worn items.
+    look_result = DRC.bput("look #{Char.name}", moon_floating_regex, "You are wearing", "You are wrapped")
     # Now try to extract just the moon weapon name from what we saw.
     scan_result = look_result.scan(moon_weapon_regex)
     # If a match was found then grab it from the captured groups per the regex.
     if scan_result.length > 0 && scan_result[0].length > 0
       moon_weapon = scan_result[0][0]
       held_it = DRC.bput("hold #{moon_weapon}", *moon_hold_messages) == "You grab"
+    else
+      # At this point you're neither holding a moon weapon nor did we see one
+      # hovering around you. In case you're wearing a shadowsilk cloak or other
+      # item that hides all your features then we will brute force try to hold
+      # all known moon weapon types as a last ditch effort.
+      held_it = held_it || DRC.bput("hold moonblade", *moon_hold_messages) == "You grab"
+      held_it = held_it || DRC.bput("hold moonstaff", *moon_hold_messages) == "You grab"
```

Comment:
Handle if you're wrapped in a shadowsilk cloak and we can't see which moon weapon you may be wearing.